packages/gatsby-theme-sso-ui-kit/src/components/avatar.js

```diff
@@ -0,0 +1,128 @@
+// A React component to be rendered in the top bar next to the top menu toggle button
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from '@emotion/styled';
+import { useAuth0 } from '@auth0/auth0-react';
+import Spacings from '@commercetools-uikit/spacings';
+import { designSystem } from '@commercetools-docs/ui-kit';
+import { GraduationCapIcon } from '@commercetools-uikit/icons';
+import LoginButton from './login-button';
+import LogoutButton from './logout-button';
+import PrimaryButton from './primary-button';
+import { getAvatarInitials } from './sso.utils';
+import { AUTH0_CLAIM_DISPLAYNAME } from '../sso.const';
+
+const AvatarContainer = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  padding-right: ${designSystem.dimensions.spacings.s};
+`;
+
+const Avatar = styled.div`
+  width: ${designSystem.typography.lineHeights.cardSmallTitle};
+  height: ${designSystem.typography.lineHeights.cardSmallTitle};
+  color: white;
+  background-color: ${designSystem.colors.light.link};
+  border-radius: 50%;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+`;
+
+const UserAvatar = (props) => {
+  const email = props.userData.find(
+    (item) => Object.keys(item)[0] === 'email'
+  )?.email;
+  const name = props.userData.find(
+    (item) => Object.keys(item)[0] === 'name'
+  )?.name;
+  const avatarInitials = getAvatarInitials(name || email).map((initial) =>
+    initial.toUpperCase()
+  );
+  return (
+    <Spacings.Inline alignItems="center">
+      <Avatar>{avatarInitials}</Avatar>
+    </Spacings.Inline>
+  );
+};
+UserAvatar.displayName = 'UserAvatar';
+
+const LoggedInState = (props) => {
+  return (
+    <AvatarContainer data-test-id="avatar-container">
+      <LogoutButton />
+      <UserAvatar userData={props.userData} />
+    </AvatarContainer>
+  );
+};
+
+LoggedInState.displayName = 'LoggedInState';
+
+const LoggedOutState = () => {
+  const { loginWithRedirect } = useAuth0();
+
+  return (
+    <AvatarContainer>
+      <LoginButton data-test-id="login-button" label="Login" />
+      <PrimaryButton
+        onClick={() =>
+          loginWithRedirect({
+            authorizationParams: {
```

Comment:
@timonrey 
I think you need here the appState and returnTo parameter as in the login button .. otherwise the user will not be returned to the right page after signup